src/components/data-display/charts/RNPChart.tsx

```diff
@@ -0,0 +1,283 @@
+import { AoGetCostDetailsParams, mARIOToken } from '@ar.io/sdk';
+import { useCostDetails } from '@src/hooks/useCostDetails';
+import { useGlobalState, useWalletState } from '@src/state';
+import { formatARIOWithCommas, formatDateMDY } from '@src/utils';
+import { useEffect, useRef, useState } from 'react';
+import {
+  Line,
+  LineChart,
+  ResponsiveContainer,
+  Tooltip,
+  XAxis,
+  YAxis,
+} from 'recharts';
+import { CategoricalChartState } from 'recharts/types/chart/types';
+import { Coordinate } from 'recharts/types/util/types';
+
+const START_RNP_PREMIUM = 50;
+
+export function RNPChart({
+  name,
+  purchaseDetails,
+}: {
+  name: string;
+  purchaseDetails?: AoGetCostDetailsParams;
+}) {
+  const [{ walletAddress }] = useWalletState();
+  const { data: costDetails } = useCostDetails({
+    intent: 'Buy-Record',
+    years: 1,
+    type: 'lease',
+    fromAddress: walletAddress?.toString(),
+    name,
+    ...(purchaseDetails ?? {}),
+  });
+
+  type ChartData = {
+    timestamp: number;
+    price: number;
+  }[];
+
+  const [chartData, setChartData] = useState<ChartData>([]);
+  const [activePayload, setActivePayload] = useState<any>(null);
+  const [defaultTooltip, setDefaultTooltip] = useState<any>(null);
+  const [defaultTooltipCoords, setDefaultTooltipCoords] =
+    useState<Partial<Coordinate> | null>(null);
+  const [defaultTooltipIndex, setDefaultTooltipIndex] = useState<any>(0);
+  const [tooltipData, setTooltipData] = useState<any>(null);
+
+  const chartRef = useRef<{ state: CategoricalChartState }>(null);
+  const [refReady, setRefReady] = useState(false);
+
+  // NOTE: there is *SOMETHING* wrong with chartRef in that it only *sometimes* triggers the below effect that positions the tooltip.
+  // moving that effect up in the order helps some, but not perfectly. This was the only way to ensure reliable (if slow) triggering of that effect to set the position.
+
+  // TODO: get rid of this nasty garbage and find a better solution.
```

Comment:
Yes this isn't reliable. ref.current doesn't trigger re-renders and shouldn't be generally used like this. Add a task to JIRA to fix this so it doesn't get lost.